sonaranalyzer-dotnet/its/regression-test.ps1

```diff
@@ -25,33 +35,121 @@ function Build-Project([string]$ProjectName, [string]$SolutionRelativePath) {
-    Copy-Item ".\AllSonarAnalyzerRules.ruleset" -Destination ".\output"
+
+    if ($ruleId) {
+        $template = Get-Content -Path ".\SingleRule.ruleset.template" -Raw
+        $rulesetWithOneRule = $template.Replace("<Rule Id=`"$ruleId`" Action=`"None`" />", `
+                                                "<Rule Id=`"$ruleId`" Action=`"Warning`" />")
+        Set-Content -Path ".\output\AllSonarAnalyzerRules.ruleset" -Value $rulesetWithOneRule
+    }
+    else {
+        Copy-Item ".\AllSonarAnalyzerRules.ruleset" -Destination ".\output"
+    }
+    Write-Host "Initialize-OutputFolder:" $timer.Elapsed.TotalSeconds
+}
+
+function Export-AnalyzerPerformancesFromLogs([string[]]$buildLogsPaths) {
+    return $buildLogsPaths |
+        Foreach-Object { Get-Content $_ } |
+        Where-Object { $_ -match '^\s*<?([0-9.]+)\s*<?[0-9]+\s*(SonarAnalyzer\..*)$' -and ($matches[1] -ne '0.001') } |
+        Foreach-Object {
+            New-Object PSObject -Property @{
+                Rule = $matches[2];
+                Time = [decimal]$matches[1]}
+        } |
+        Group-Object Rule |
+        Foreach-Object {
+            New-Object PSObject -property @{
+                Rule = $_.Name;
+                Time = [math]::Round(($_.Group.Time | Measure-Object -sum).Sum, 3)}
+        } |
+        Sort-Object Time -Descending
+}
+
+function GetFullPath($folder) {
+    return [System.IO.Path]::GetFullPath((Join-Path (Get-Location).Path $folder))
+}
+
+function CopyFolderRecursively($from, $to, $include, $exclude) {
+    $fromPath = GetFullPath -folder $from
+    $toPath = GetFullPath -folder $to
+
+    $files = if ($include)
+    {     Get-ChildItem -Path $fromPath -Recurse -Include $include }
+    else {Get-ChildItem -Path $fromPath -Recurse -Exclude $exclude }
+
+    foreach ($file in $files) {
+        $path = Join-Path $toPath $file.FullName.Substring($fromPath.length)
+        $parent = split-path $path -parent
+        if (-Not (Test-Path $parent)) {
+            New-Item $parent -Type Directory -Force | out-null
+        }
+        Copy-Item $file -Destination $path
+    }
+}
+
+function Measure-AnalyzerPerformance()
+{
+    # Process all build logs in the "output" folder
+    $timings = Export-AnalyzerPerformancesFromLogs(Get-ChildItem output -filter *.log | Foreach-Object { $_.FullName })
+
+    $timings | Where-Object { $_.Rule -match 'SonarAnalyzer\.CSharp.*' }                |
+        Format-Table -Wrap -AutoSize | Out-String -Width 100
+    $timings | Where-Object { $_.Rule -match 'SonarAnalyzer\.VisualBasic.*' }           |
+        Format-Table -Wrap -AutoSize | Out-String -Width 100
+    $timings | Where-Object { $_.Rule -match 'SonarAnalyzer\.Rules\.CSharp.*' }         |
+        Format-Table -Wrap -AutoSize | Out-String -Width 100
+    $timings | Where-Object { $_.Rule -match 'SonarAnalyzer\.Rules\.VisualBasic.*' }    |
+        Format-Table -Wrap -AutoSize | Out-String -Width 100
+}
+
+function DiffResults() {
```

Comment:
We have tried to always comply to powershell naming guideline. Please rename this method to comply to it.

Reply:
We (I mean in my past life) followed the guideline that only public/exported functions follow the Verb-Noun, and the private functions don't - so it's very easy to tell what's safe to change and what is potentially used anywhere. I'll follow that for new scripts, but will adjust this one as you suggest. Mixing two styles in one script is definitely bad.